clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -279,6 +279,10 @@ public void abort() {
         aborted = true;
     }
 
+    public void unsetProducerState() {
+        builtRecords = null;
```

Comment:
What should the state of the builder be for this? Maybe we should verify at least that it wasn't aborted?